### PR TITLE
ci: Combine all go tests into one job

### PR DESCRIPTION
**Pull request by mslipper**
Combines all Go tests into a single job. This allows the runner to decide how to best parallelize the tests, and simplifies our CCI config.

### CODE REVIEW

File: .circleci/config.yml
Comment (ajsutton) [inline .circleci/config.yml:1399]: This would be a good one to add to the list of explicitly required tasks in GitHub (maybe in a few days after PRs have had a chance to update or be merged before blocking on it).  If it's not specifically listed the merge queue takes longer to kick things out when it fails.

Although surprisingly none of the old ones were explicitly required so could just see how it goes.

**Review by ajsutton — APPROVED** [review]

File: .circleci/config.yml
Comment (mslipper) [reply .circleci/config.yml:1399]: Good idea - will do so once the current batch of PRs goes through.

**Pull request merged**
